controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -652,6 +666,134 @@ bool master_thread::handle_cmdu_1905_channel_preference_report_message(
     return son_actions::send_cmdu_to_agent(sd, cmdu_tx);
 }
 
+bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
+                                                                ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received CHANNEL_SELECTION_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
+    int tlvType;
+    while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
+
+        LOG(DEBUG) << "tlvType=" << std::hex << int(tlvType);
+
+        if (tlvType < 0) {
+            LOG(ERROR) << "getNextTlvType has failed";
+            return false;
+        }
+
+        if (tlvType != int(wfa_map::eTlvTypeMap::TLV_CHANNEL_SELECTION_RESPONSE)) {
+            LOG(ERROR) << "Unexpected tlv type in TLV_CHANNEL_SELECTION_RESPONSE, type="
+                       << int(tlvType);
+            // TODO: replace return statement with function that skip s unexpected tlv
+            // see: https://github.com/prplfoundation/prplMesh/issues/107
+            return false;
+        }
+
+        auto channel_selection_response_tlv =
+            cmdu_rx.addClass<wfa_map::tlvChannelSelectionResponse>();
+        if (!channel_selection_response_tlv) {
+            LOG(ERROR) << "addClass ieee1905_1::tlvChannelSelectionResponse has failed";
+            return false;
+        }
+
+        auto &ruid         = channel_selection_response_tlv->radio_uid();
+        auto response_code = channel_selection_response_tlv->response_code();
+
+        LOG(DEBUG)
```

Comment:
Move this to a different function which will return the string for the DEBUG print

Reply:
Why move it to a different function? It is only needed in one place and no other code will call it. 
I don't see how it is better. The lambda function is convenient and perfectly fits this.
just like (cond ? true : false) but for more complex code.